### PR TITLE
Adding support for Live Metrics in .NET Core

### PR DESCRIPTION
**Pull request by netgh0st**
Addresses the issue https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/216

### CODE REVIEW

**Comment by cijothomas** [comment]
Can you check the build failures? (the one named 'clone' runs more tests than other.)

**Comment by SergeyKanzhelev** [comment]
I think it's related to https://github.com/Microsoft/ApplicationInsights-aspnetcore/pull/517 I asked @netgh0st to wait till that PR completed

**Comment by SergeyKanzhelev** [comment]
@netgh0st please merge and enable QP for .NET Core. I updated references to the NuGets in separate PR already

**Comment by cijothomas** [comment]
Fixes: https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/460

File: src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs [hunk 1]
```diff
@@ -95,25 +95,25 @@ public void Configure(TelemetryConfiguration configuration)
 
         private void AddTelemetryChannelAndProcessorsForFullFramework(TelemetryConfiguration configuration)
         {
+            // Enabling Quick Pulse Metric Stream
+            if (this.applicationInsightsServiceOptions.EnableQuickPulseMetricStream)
```
Comment (SergeyKanzhelev) [inline src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs]: so it is a change in behavior. For full framework fro some reasons we didn't enable QP for non server channel. Perhaps make the same change for sampling then

File: src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 2]
```diff
@@ -65,14 +65,14 @@
...
+    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.5.0-beta1-build10212" />
+    <PackageReference Include="Microsoft.ApplicationInsights.DependencyCollector" Version="2.5.0-beta1-build01010" />
+    <PackageReference Include="Microsoft.ApplicationInsights.PerfCounterCollector" Version="2.5.0-beta1-build01010" />
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'net451' OR '$(TargetFramework)' == 'net46' ">
     <PackageReference Include="System.Net.Http" Version="4.3.1" />
-    <PackageReference Include="Microsoft.ApplicationInsights.PerfCounterCollector" Version="2.4.1" />
-    <PackageReference Include="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" Version="2.4.0" />
+    <PackageReference Include="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" Version="2.5.0-beta1-build10212" />
```
Comment (SergeyKanzhelev) [inline src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj]: Please merge with develop. File was already changed

File: src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs (hunk 1 above)
Comment (netgh0st) [reply src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptionsSetup.cs]: Done

**Comment by netgh0st** [comment]
Build succeeded on re-run https://mseng.visualstudio.com/AppInsights/_build/index?buildId=4231537&_a=summary

**Comment by chunliu** [comment]
what is the status regarding to the beta version?

**Comment by cijothomas** [comment]
@chunliu - beta1 for asp.net core sdk is waiting for the Application Insights Web SDK to be released as this has a dependency. Web SDK is undergoing final testing and should be out in 2-3 days. As soon as its ready, we'll publish asp.net core sdk as well.
This release has been delayed due to some infra issues, thanks for your patience.

**Comment by kbdavis07** [comment]
Any updates on this?

**Pull request merged**